### PR TITLE
GiniSwitchSdk refactoring

### PR DESCRIPTION
**Pull request by nsobadjiev**
# Introduction

There was a problem with the old architecture that `ExtractionsManager` - the class responsible for interacting with the backend, was created and used by the UI classes in the SDK. That meant that the UI had to stay alive even if it wasn't on screen. Also, `GiniSwitchSdk ` had no purpose other than holding the credentials, which was making things difficult at times.

# How to test

Run the tests and start the demo app - everything should work as usual

# Merge info

Automatic

### CODE REVIEW

File: GiniSwitchSDK/Example/Tests/MultiPageCoordinatorTests.swift [hunk 1]
```diff
@@ -47,42 +46,65 @@ class MultiPageCoordinatorTests: XCTestCase {
...
+//    func testReviewControllerDismiss() {
+//        coordinator.cameraController.takePicture()
+//        requestedReviewScreen?.rejectButtonTapped()
+//        XCTAssertTrue(didRequestReviewDismiss, "After the image is rejected, the review screen should be dismissed")
+//    }
+//    
+//    func testExitSDKActionsSheet() {
+//        coordinator.pageCollectionController.onOptionsTapped()
+//        XCTAssertNotNil(exitActionSheet, "Tapping on the options button should result in an action sheet confirming exiting being shown")
+//    }
```
Comment (EarlOfEgo) [inline GiniSwitchSDK/Example/Tests/MultiPageCoordinatorTests.swift]: I know which tests will never be rewritten. Just delete them.

File: GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchLogger.swift [hunk 2]
```diff
@@ -24,3 +24,9 @@ public protocol GiniSwitchLogger {
     func logError(message:String)
     
 }
+
+var logger:GiniSwitchLogger = GiniSwitchConsoleLogger()
+
+func Logger() -> GiniSwitchLogger {
```
Comment (EarlOfEgo) [inline GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchLogger.swift]: I think it is weird to have a function start with an upper case letter. Why not `getLogger()` and why is it even needed can't I just access `logger`?

File: GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift [hunk 3]
```diff
@@ -56,26 +48,72 @@ public class GiniSwitchSdk {
...
+        extractionsManager.sendFeedback(feedback)
+    }
+
+}
+
+extension GiniSwitchSdk: ExtractionsManagerDelegate {
+    
+    func extractionsManager(_ manager:ExtractionsManager, didEncounterError error:NSError) {
+        // check if it's an error that needs to be shown to the user
+        if error.isHumanReadable {
```
Comment (EarlOfEgo) [inline GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift:74]: The comment and the name of the method are two different things. The comment suggest if there is an error that should be shown and the function name says if an error is readable. Comments are bad.

File: GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift [hunk 4]
```diff
@@ -56,26 +48,72 @@ public class GiniSwitchSdk {
...
+        
+    }
+    
+    func extractionsManagerDidCreateOrder(_ manager:ExtractionsManager) {
+        
+    }
+    
+    func extractionsManagerDidCompleteExtractions(_ manager:ExtractionsManager) {
+        // Don't interrupt the user immediately. Allow them to complete the last action
+        // before notifying them that extractions are complete
```
Comment (EarlOfEgo) [inline GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift:99]: is the actual delay happening here? I don't get this.

**Review by EarlOfEgo — CHANGES_REQUESTED** [review]
Just some questions, suggestions.
And maybe adjust something in the documentation about  that the custom attributes are now static.

File: GiniSwitchSDK/Example/Tests/MultiPageCoordinatorTests.swift (hunk 1 above)
Comment (nsobadjiev) [reply GiniSwitchSDK/Example/Tests/MultiPageCoordinatorTests.swift]: You know me so well...

File: GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift (hunk 3 above)
Comment (nsobadjiev) [reply GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift:74]: Why? If the error is human readable, it should be shown. If it's not, users will not be able to make sense of it, so we shouldn't show it at all.

File: GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift (hunk 4 above)
Comment (nsobadjiev) [reply GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift:99]: It's this thing that the "Extractions Complete" (Splash) screen is only shown after the review screen. The thing you don't have on Android. So if we see that all extractions are there, we just mark them so and wait for the user to take another pic.

**Review by EarlOfEgo — APPROVED** [review]

**Pull request merged**